src/hotspot/share/gc/shenandoah/c1/shenandoahBarrierSetC1.cpp

```
@@ -296,7 +298,12 @@ void ShenandoahBarrierSetC1::generate_c1_runtime_stubs(BufferBlob* buffer_blob)
     _load_reference_barrier_phantom_rt_code_blob = Runtime1::generate_blob(buffer_blob, C1StubId::NO_STUBID,
                                                                            "shenandoah_load_reference_barrier_phantom_slow",
                                                                            false, &lrb_phantom_code_gen_cl);
+    reference_barrier_success = _load_reference_barrier_strong_rt_code_blob != nullptr &&
+                                _load_reference_barrier_strong_native_rt_code_blob != nullptr &&
+                                _load_reference_barrier_weak_rt_code_blob != nullptr &&
+                                _load_reference_barrier_phantom_rt_code_blob != nullptr;
   }
+  return _pre_barrier_c1_runtime_code_blob != nullptr && reference_barrier_success;
```

Wouldn't it be better to return false immediately after each failure, rather than continuing?

---

Yes, totally. Fixed.